### PR TITLE
fix(eslint): Add freeCodeCamp config

### LINKED ISSUE #232: Having FCC Eslint Rules Be Importable
In Issue [120](https://github.com/freeCodeCamp/testable-projects-fcc/issues/120) of testable-projects-fcc, I was discussing having an importable eslint config. This would require a separate repository to be set up for it.

I created a repository here https://github.com/jnmorse/eslint-config-fcc that does include the fcc rules and can be installed as a dev package to serve as an example. This package is using a more up to date release of eslint 4.7.2.

Think it would be best if the package that is used was published to npm so it could be installed like a normal package.

At @tchaffee suggestion, I am opening the issue and will post in the main repo as well for discussion.

The benefits to having eslint-config package are that it can be maintained and used for multiple projects, and campers could include it as a dependency for their own projects. The downside is that would be external and all changes will have to happen there, aside from any project specific rules.

### PR DESCRIPTION
**Pull request by jnmorse**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file
- [x] Working changes have been added to the build by running `npm run build`

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue: Closes #232 

#### Description

Added the freeCodeCamp eslint package, and added a couple
eslint-disable-next-lines in test files.

### CODE REVIEW

File: test/automate/automate-utils.js [hunk 1]
```diff
@@ -20,6 +20,8 @@
  * no longer need Xvfb in the travis.yml file.
  */
 
+// no idea why this triggers.
+// eslint-disable-next-line import/unambiguous
 let screenShotUniq = 0;
```
Comment (tbushman) [inline test/automate/automate-utils.js]: Thanks @jnmorse . The documentation on unambiguous has broken links, so I tried updating all the eslint dev dependencies, and that resulted in the whole file being reported, not just the variable. I was able to suppress the warning by removing `"sourceType": "module"` from .eslintrc. Do you think it's ok not to have that parserOption ?

Comment (jnmorse) [reply test/automate/automate-utils.js]: When I first added the package and tried it locally without the sourceType eslint complains about the import statements. Airbnb config actually has this rule turned off completely.

this is the comment on the rule in their config

```
// Warn if a module could be mistakenly parsed as a script by a consumer
    // leveraging Unambiguous JavaScript Grammar
    // https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/unambiguous.md
    // this should not be enabled until this proposal has at least been *presented* to TC39.
    // At the moment, it's not a thing.
```

I'm not even clear on what this rule is supposed to be about.

Comment (jnmorse) [reply test/automate/automate-utils.js]: Looks like they will pass without it if I update the eslint package to latest. Will re-push in a moment.

**Comment by tbushman** [comment]
Regarding failed check, the serve-https docs mention this: `# BEFORE you access in a browser for the first time, use curl (because there's a concurrency bug in the greenlock setup)
curl https://jane.daplie.me`  
Unsure whether that applies to this project... the https://localhost.daplie.me:8080 server works locally, but perhaps only because it isn't the first time running it, whereas the travis server is rebuilt each time, and unsure why it's happening now. Guessing ...

**Comment by tchaffee** [comment]
It looks to me like Daplie is having some problems with their packages. I have reached out to them asking for a fix.

File: package.json [hunk 2]
```diff
@@ -28,7 +29,8 @@
   },
   "scripts": {
     "serve-https-local": "serve-https -p 8080",
-    "pretest": "eslint src/ test/",
+    "lint": "eslint src/ test/",
+    "pretest": "npm run lint",
```
Comment (tchaffee) [inline package.json]: I don't quite understand the reason for these changes? What benefit does the extra line of config give us?

File: test/codepen-projects.js [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 // Mocha or Selenium is creating more than the max default of 10 events
 // emitters, so we increase the default max here.
+
```
Comment (tchaffee) [inline test/codepen-projects.js]: We usually don't put a blank line between inline comments and code. But if we did want to reformat this file, does it really belong in this PR?

Comment (jnmorse) [reply test/codepen-projects.js]: I think I actually included this blank line by accident. I can certainly remove it and resubmit it without the blank line.

File: package.json (hunk 2 above)
Comment (jnmorse) [reply package.json]: Perhaps this is partly a personal reflection, I'm used to most of the projects I've worked with having a lint script. I tend to run linting from the command line on a fairly regular basis. I had made the change when I was testing to make sure the config worked like it was supposed to.

I think it makes a bit more sense then runing `npm run pretest` from the command line to only run eslint.

I will remove it if you don't feel it's appropriate addition.

Comment (tchaffee) [reply package.json]: Totally makes sense, and now I can see why it's useful. Thanks for taking the time to explain. Let's keep this.

File: test/codepen-projects.js (hunk 3 above)
Comment (tchaffee) [reply test/codepen-projects.js]: Yes, this file has nothing to do with the PR, so let's get rid of this change. I'll be happy to merge this PR once this change is removed.

**Comment by tchaffee** [comment]
@jnmorse Everything looks good, I'm going ahead with the merge. Thanks for your work on this.

**Pull request merged**